aws/retry/middleware.go

```diff
@@ -328,10 +328,19 @@ func AddRetryMiddlewares(stack *smithymiddle.Stack, options AddRetryMiddlewaresO
 		middleware.LogAttempts = options.LogRetryAttempts
 	})
 
-	if err := stack.Finalize.Add(attempt, smithymiddle.After); err != nil {
-		return err
+	// index retry to before signing, if signing exists
+	_, ok := stack.Finalize.Get("Signing")
```

Comment:
We shouldn't do this conditionally, `Signing` should always be there, and if it wasn't it should be something we fail on rather than ignore silently.

Reply:
it was failing on IMDS calls. specifically `GetMetadata`. looking at its implementation, it doesnt appear to add signing middleware